src/color-extractor.ts

```diff
@@ -0,0 +1,364 @@
+const { rgb2lab, deltaE } = require('rgb-lab');
+
+
+/**
+ * Takes in an ImageData object and finds the k most dominant colors from
+ * the image data. Also returns the number of pixels assoiated with each 
+ * color's cluster. Dominant colors that are too close to colors in 
+ * colorsToIgnore are ignored.
+ * 
+ * @param myData          ImageData generated from a HTML Canvas element
+ * @param numClusters     Number of dominant colors to find
+ * @param colorsToIgnore  Array of colors as RGB strings to ignore
```

Comment:
RE: RGB strings. Plz document the acceptable string format, and optionally provide an example string.